File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpRespawnData.java
package calebxzhou.rdi.mc.common2.mcp;

public record RMcpRespawnData(
        boolean wasDead,
        boolean respawned,
        boolean hardcore,
        float beforeHealth,
        float afterHealth,
        RMcpPosData before,
        RMcpPosData after
) {
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpRespawnData.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpRespawnData(
    val wasDead: Boolean,
    val respawned: Boolean,
    val hardcore: Boolean,
    val beforeHealth: Float,
    val afterHealth: Float,
    val before: REntityPosData,
    val after: REntityPosData?
)
